cmd/tpm-manager/tpm-manager.go

```diff
@@ -0,0 +1,506 @@
+		invalidateNode(node)
+		return nil
+	}
+
+	err = tpm.ValidateLog(log, quote)
+	if err != nil {
+		glog.Errorf("TPM event log does not match quote for node %s", address.String())
+		invalidateNode(node)
+		return nil
+	}
+
+	logstate, err := tpm.ValidatePCRs(log, quote, manager.pcrconfigs)
+	jsonlog, _ := json.Marshal(logstate)
+	if node.ObjectMeta.Annotations == nil {
+		node.ObjectMeta.Annotations = make(map[string]string)
+	}
+	node.ObjectMeta.Annotations["tpm.coreos.com/logstate"] = string(jsonlog)
+	if err != nil {
+		glog.Errorf("TPM quote PCRs don't validate for node %s", address.String())
+		invalidateNode(node)
+		return nil
+	}
+
+	glog.Errorf("TPM quote for node %s validates", address.String())
+	validateNode(node)
+	return nil
+}
+
+func verifyAndUpdate(node *api.Node) {
+	verifyNode(node)
+
+	// The state that the node will be updated to
+	newstate := isTrusted(node)
+	newnode, err := manager.client.Nodes().Get(node.Name)
+	if err != nil {
+		glog.Errorf("Unable to obtain node state for %s: %v", node.Name, err)
+		return
+	}
+	if newnode.ObjectMeta.Annotations == nil {
+		newnode.ObjectMeta.Annotations = make(map[string]string)
+	}
+	currenttime := time.Now().Unix()
+
+	// If we're transitioning state, update the metadata
+	if isTrusted(newnode) != newstate {
+		if newstate == false {
+			newnode.ObjectMeta.Annotations["tpm.coreos.com/untrustedsince"] = strconv.FormatInt(currenttime, 10)
+			newnode.ObjectMeta.Annotations["tpm.coreos.com/trustedsince"] = ""
+		} else {
+			newnode.ObjectMeta.Annotations["tpm.coreos.com/untrustedsince"] = ""
+			newnode.ObjectMeta.Annotations["tpm.coreos.com/trustedsince"] = strconv.FormatInt(currenttime, 10)
+		}
+	}
+
+	newnode.ObjectMeta.Annotations["tpm.coreos.com/validationtime"] = strconv.FormatInt(currenttime, 10)
+
+	// Ensure that the new node is tainted appropriately
+	if newstate == true {
+		validateNode(newnode)
+	} else {
+		invalidateNode(newnode)
+	}
+	newnode.ObjectMeta.Annotations["tpm.coreos.com/logstate"] = node.ObjectMeta.Annotations["tpm.coreos.com/logstate"]
+	newnode, err = manager.client.Nodes().Update(newnode)
+	if err != nil {
+		glog.Errorf("Unable to update node state for %s: %v", node.Name, err)
+		return
+	}
+}
+
+func verifyAllNodes() {
+	nodes, err := manager.client.Nodes().List(api.ListOptions{})
+	if err != nil {
+		return
+	}
+	for _, node := range nodes.Items {
+		verifyAndUpdate(&node)
+	}
+}
+
+func reverify() {
+	select {
+	case <-time.After(time.Duration(manager.recurring) * time.Second):
+		verifyAllNodes()
+	case <-manager.recurringChan:
+	}
+}
+
+type TPMManager struct {
+	Master        string
+	Kubeconfig    string
+	tpmhandler    tpm.TPMHandler
+	pcrconfig     string
+	pcrconfigdir  string
+	allowunknown  bool
+	recurring     int
+	client        *client.Client
+	policyTimer   *time.Timer
+	leaderelect   componentconfig.LeaderElectionConfiguration
+	recurringChan chan int
+	pcrconfigs    []map[string]tpm.PCRConfig
+}
+
+var manager TPMManager
```

Comment:
nit: relying on global state for something this featured feels a bit off. I don't have any concrete problems with this as it stands now, but it could make reasoning about the use a little difficult